### DISCUSSION TITLE
Performance issue between maui and xamarin - liteDB

### DISCUSSION THREAD

**Discussion by LukasKuchta** [dotnet/maui#12743] [General]
Hello, we are porting Xamarin app into Maui/.net7 and we have issue with performance. Code is same "only" platfrom is different. 

Db has 30K records. 
xamarin - read all => 20s
maui - read all -> 80s 
windows - xUnit test -> super fast

I tested it on Android. iOS will be later. 

What can be problem with it any .net7 features ... ? Or any advice for debuging this behaviour ?

**Comment by xtuzy** [comment]
they are all test at release? net7.0-android still use mono, i don't think it's performance is lower than xamarin.android, you can create a simple sample to test, if it is lower, create issue at xamarin-android or litedb, not maui

**Reply by LukasKuchta** [reply]
@xtuzy 

We made small sample apps. Empty no unnecessary code here. 

And result is: 
MAUI (net7.0-ios):
 Upserted 25000 items in 50.0988223
 Selected 25000 items as TeamItemDto in 30.6884797
 Selected 25000 items as Bson in 14.0719025

 Xamarin.iOS:
 Upserted 25000 items in 32.9143934
 Selected 25000 items as TeamItemDto in 6.9651505
 Selected 25000 items as Bson in 4.4160259

Can you someone explain me what is wrong? Any optimization settings? It seems MAUI problem no liteDB.

**Reply by xtuzy** [reply]
where i can see code?

**Comment by LukasKuchta** [comment]
@xtuzy there is the source code: https://github.com/LukasKuchta/XamarinMauiDbTest simple maui and xamarin app. 

and there are some numbers
https://github.com/LukasKuchta/XamarinMauiDbTest/wiki

We added the SQLite for comparsion and issue in performance is visible too. Seems no db engine problem.

MAUI performnance is very bad against the xamarin. Thanks for check.

**Reply by xtuzy** [reply]
see this issue: https://github.com/dotnet/maui/issues/11479
